src/json-api/json-api-hapi-plugin.ts

```diff
@@ -589,20 +640,69 @@ export class JsonApiHapiPlugin {
     return reply(this.getEntity('notification', api => api.getProjectNotificationConfigurations(projectId)));
   }
 
-  public async postNotificationConfigurationHandler(request: Hapi.Request, reply: Hapi.IReply) {
-    const config = request.payload.data.attributes;
-
-    config.teamId = config.teamId || null;
-    config.projectId = config.projectId || null;
+  private async authorizeNotificationConfiguration(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      reply(request.payload.data.attributes);
+    } catch (error) {
+      reply(Boom.badRequest());
+    }
+  }
 
-    if (!config.teamId && !config.projectId) {
-      throw Boom.badRequest('teamId or projectId should be defined');
+  private async authorizeTeamOrProjectAccess(request: Hapi.Request, reply: Hapi.IReply) {
+    try {
+      const config = request.pre.filter;
```

Comment:
Yep.

The reason I hoped to change the method name would be to provide a better hint on what the implicit contract of this function is. 

But a better way than fiddling with the function name would be to make the contract explicit by rewriting the function as follows:
```typescript
  private async authorizeTeamOrProjectAccess(
    request: {
      pre: { filter: { teamId: number, projectId: number }},
      userHasAccessToProject: (projectId: number) => boolean,
      userHasAccessToTeam: (teamId: number) => boolean,
    },
    reply: Hapi.IReply,
  ) {
    try {
      const { teamId, projectId } = request.pre.filter;

      if (!teamId && !projectId) {
        return reply(Boom.badRequest('teamId or projectId should be defined'));
      }
      if (teamId && projectId) {
        return reply(Boom.badRequest('teamId and projectId should not both be defined'));
      }
      if (projectId) {
        if (await request.userHasAccessToProject(projectId)) {
          return reply(request.pre.filter);
        }
      }
      if (teamId) {
        if (await request.userHasAccessToTeam(teamId)) {
          return reply(request.pre.filter);
        }
      }
    } catch (exception) {
      // TODO: log exception
    }
    return reply(Boom.unauthorized());
  }
```